### ISSUE TITLE
Improve the `smv-release` process to support new release branch

### ISSUE THREAD

**Issue by cloudysail** [TresAmigosSD/SMV#1114]
Right now when doing `smv-release` for a new branch(e.g. SMV branch 2.2 has never been released before), there is the issue that we are using docker image of PREV_SMV_VERSION(2.1.x) which contains the wrong spark version. 

Since we only rely on the spark rather than SMV in the docker image to do `smv-release`, we can add one step at the beginning of `smv-release`, which is to build a first docker image(at the end we still need to build the seconds docker image which is as it is now) for the right spark version in current SMV. We can improve the `spark-install` to support spark version as parameter, and reuse it when building the first docker image.

**Comment by cloudysail** [comment]
As mentioned in the description, I added one step to build a first docker image(local image) before doing sbt assembly and test. 

To have the right spark version in it, I had a quick solution:
Each tools/smv_install use the corresponding spark download link starting at 2.1. For example,2.1 smv_install uses spark 2.1.1, and 2.2 smv_install uses spark 2.2.0, etc.. Then each dockFile calls the corresponding smv_install in its branch rather than the previous_smv_version.

Now the V2.2.0.0 has been released except the very last step - It seems my user does not have permission to push new image to docker hub.
